AsyncDisplayKit/ASMapNode.mm

```diff
@@ -150,23 +146,33 @@ - (void)takeSnapshot
   }
 }
 
+- (void)setUpSnapshotter
+{
+    if (!_snapshotter) {
+        ASDisplayNodeAssert(!CGSizeEqualToSize(CGSizeZero, self.calculatedSize), @"self.calculatedSize can not be zero. Make sure that you are setting a preferredFrameSize or wrapping ASMapNode in a ASRatioLayoutSpec or similar.");
+        _options.size = self.calculatedSize;
+        _snapshotter = [[MKMapSnapshotter alloc] initWithOptions:_options];
+    }
+}
+
 - (void)resetSnapshotter
 {
   if (!_snapshotter.isLoading) {
-    _options.size = self.calculatedSize;
     _snapshotter = [[MKMapSnapshotter alloc] initWithOptions:_options];
   }
 }
 
-#pragma mark - Action
+#pragma mark - Actions
 - (void)addLiveMap
 {
   if (self.isNodeLoaded && !_mapNode) {
-    _mapNode = [[ASDisplayNode alloc]initWithViewBlock:^UIView *{
-      _mapView = [[MKMapView alloc]initWithFrame:CGRectMake(0.0f, 0.0f, self.calculatedSize.width, self.calculatedSize.height)];
+    __weak ASMapNode *weakSelf = self;
+    _mapNode = [[ASDisplayNode alloc] initWithViewBlock:^UIView *{
+      _mapView = [[MKMapView alloc] initWithFrame:CGRectZero];
       _mapView.delegate = _mapDelegate;
       [_mapView setRegion:_options.region];
```

Comment:
@appleguy I've ran this through the kittens sample app and it doesn't seem to cause a retain cycle (not one I can spot anyway) Although this isn't an area of expertise for myself. I'd be happy to make all those ivars have a weak local reference but it doesn't look clean visual. Is this the route you would propose? 

In terms of _mapNode. I've decided to kill it and work with _mapView exclusively as you've suggested. Works better this way. I've also added that assert to the top